File: backend-api/Entities/Media.cs
namespace backend_api.Entities;

public class Media
{
    public Guid Id { get; set; }

    public string Name { get; set; } = string.Empty;

    public string Description { get; set; } = string.Empty;

    public MediaType Type { get; set; }

    public string FileName { get; set; } = string.Empty;

    public DateTime CreatedAt { get; set; }
}

File: web-player/src/types/media.ts
type MediaType = "image" | "video";

type Media = {
  id: string;
  name: string;
  description: string;
  type: MediaType;
  url: string;
  createdAt: string;
};

type MediaResponse = {
  id: string;
  name: string;
  description: string;
  type: MediaType;
  url: string;
  createdAt: string;
};

export type { MediaType, Media, MediaResponse };
